### PR TITLE
chore: callback method in getDatabaseDialect

### LINKED ISSUE #2075: Database.getDatabaseDialect's "cb" argument is never used
If we examine the method "getDatabaseDialect" https://github.com/googleapis/nodejs-spanner/blob/30151cad2723d6a9f2c70342c81f1a578ee02f6a/src/database.ts#L1530-L1553
we can see that the callback passed in is never used. I found this while working on trace instrumentation for these libraries.

### PR DESCRIPTION
**Pull request by alkatrivedi**
fixes: #2075

### CODE REVIEW

File: src/database.ts [hunk 1]
```diff
@@ -1549,6 +1556,10 @@ class Database extends common.GrpcServiceObject {
       const [metadata] = await this.getMetadata(gaxOptions);
       this.databaseDialect = metadata.databaseDialect;
     }
+    if (cb) {
+      cb!(null, this.databaseDialect);
```
Comment (odeke-em) [inline src/database.ts:1560]: At this point do we still need `cb!` given that we confirmed above that the callback is non-nil?

Comment (alkatrivedi) [reply src/database.ts]: Yeah, we don't need. Updated

**Review by odeke-em — APPROVED** [review]
Thank you @alkatrivedi!

**Comment by alkatrivedi** [comment]
Thanks for approving the PR @odeke-em

**Comment by surbhigarg92** [comment]
@alkatrivedi Please add a unit test case .

File: src/database.ts [hunk 2]
```diff
@@ -1549,6 +1556,10 @@ class Database extends common.GrpcServiceObject {
       const [metadata] = await this.getMetadata(gaxOptions);
       this.databaseDialect = metadata.databaseDialect;
     }
```
Comment (surbhigarg92) [inline src/database.ts:1558]:
```suggestion
    }
    
```

Comment (alkatrivedi) [reply src/database.ts]: Updated

**Comment by alkatrivedi** [comment]
> @alkatrivedi Please add a unit test case .

A unit test already exists for this

**Review by surbhigarg92 — APPROVED** [review]

**Pull request merged**